### PR TITLE
Added User, dao, service and their implementation and authentication service with hash util

### PR DESCRIPTION
**Pull request by vicire**

### CODE REVIEW

File: src/main/java/cinema/model/Movie.java [hunk 1]
```diff
@@ -5,8 +5,10 @@
 import javax.persistence.GeneratedValue;
 import javax.persistence.GenerationType;
 import javax.persistence.Id;
+import javax.persistence.Table;
 
 @Entity
+@Table(name = "movie")
```
Comment (Max-Pochepets) [inline src/main/java/cinema/model/Movie.java]: If you specify table names, make it plural

File: src/main/java/cinema/security/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        Optional<User> foundedUser = userService.findByEmail(email);
+        if (foundedUser.isEmpty()) {
+            throw new AuthenticationException("User is not found");
+        }
+        User user = foundedUser.get();
+        String passwordToCheck = HashUtil.hashPassword(password, user.getSalt());
+        if (!user.getPassword().equals(passwordToCheck)) {
+            throw new AuthenticationException("Your login or password are incorrect. "
+                    + "Please, try again");
+        }
```
Comment (Max-Pochepets) [inline src/main/java/cinema/security/AuthenticationServiceImpl.java]: combine these two checks, do not tell what did user actually input wrong

File: src/main/java/cinema/security/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        return user;
+    }
+
+    @Override
+    public User register(String email, String password) {
+        User user = new User();
+        user.setEmail(email);
+        byte[] salt = HashUtil.getSalt();
+        String hashPassword = HashUtil.hashPassword(password, salt);
```
Comment (Max-Pochepets) [inline src/main/java/cinema/security/AuthenticationServiceImpl.java]:
```suggestion
        String hashedPassword = HashUtil.hashPassword(password, salt);
```

**Review by Max-Pochepets — CHANGES_REQUESTED** [review]

File: src/main/java/cinema/model/Movie.java (hunk 1 above)
Comment (vicire) [reply src/main/java/cinema/model/Movie.java]: Fixed

File: src/main/java/cinema/security/AuthenticationServiceImpl.java (hunk 2 above)
Comment (vicire) [reply src/main/java/cinema/security/AuthenticationServiceImpl.java]: fixed

File: src/main/java/cinema/security/AuthenticationServiceImpl.java (hunk 3 above)
Comment (vicire) [reply src/main/java/cinema/security/AuthenticationServiceImpl.java]: Fixed

**Review by Max-Pochepets — APPROVED** [review]
👍

File: src/main/java/cinema/model/User.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "clients")
```
Comment (dmytro-hryhoruk) [inline src/main/java/cinema/model/User.java]:
```suggestion
@Table(name = "users")
```

File: src/main/java/cinema/security/AuthenticationServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        }
+        return foundedUser.get();
+    }
+
+    @Override
+    public User register(String email, String password) {
+        User user = new User();
+        user.setEmail(email);
+        byte[] salt = HashUtil.getSalt();
+        String hashedPassword = HashUtil.hashPassword(password, salt);
```
Comment (dmytro-hryhoruk) [inline src/main/java/cinema/security/AuthenticationServiceImpl.java]: checklist

**Review by dmytro-hryhoruk — CHANGES_REQUESTED** [review]

File: src/main/java/cinema/model/User.java (hunk 4 above)
Comment (vicire) [reply src/main/java/cinema/model/User.java]: Changed

File: src/main/java/cinema/security/AuthenticationServiceImpl.java (hunk 5 above)
Comment (vicire) [reply src/main/java/cinema/security/AuthenticationServiceImpl.java]: Moved

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**